jormungandr/src/topology/topology.rs

```diff
@@ -226,6 +238,7 @@ impl P2pTopology {
                 if let Some(node) = &node {
                     tracing::debug!(node = %node.address(), id=?node.id(), "lifting node from quarantine");
                     self.topology.promote_peer(&node.id());
+                    self.stats_counter.sub_peer_quarantined_cnt(1);
```

Comment:
Just noting that this counter might be lagging behind the real count of quarantined nodes because evictions may happen from the lru cache and we are not able to intercept those

Reply:
We do not even have tests that account for such case in the first place, but that's a good thing to fix while we are on it.